Core/src/IfcGeometryConverter/CurveConverter.h

```diff
@@ -1342,39 +1339,296 @@ namespace OpenInfraPlatform
+
+					for (int i = 0; i < basisCurvePoints.size() - 1; i++) {
+						carve::geom::vector<3> startPoint = basisCurvePoints[i];
+						carve::geom::vector<3> endPoint = basisCurvePoints[i + 1];
+						carve::geom::vector<3> offsetPoint;
+
+						if (this->GeomSettings()->areEqual(basisCurveDirection.z, 0.0)) {
+							carve::geom::vector<2> startPoint2D = carve::geom::VECTOR(startPoint.x, startPoint.y);
+							carve::geom::vector<2> endPoint2D = carve::geom::VECTOR(endPoint.x, endPoint.y);
+
+							carve::geom::vector<2> tangent = endPoint2D - startPoint2D;
+							carve::geom::vector<2> orthogonal = carve::geom::VECTOR(-tangent.y, tangent.x);
+
+							offsetPoint = carve::geom::VECTOR(startPoint.x + distance * orthogonal.x,
+								startPoint.y + distance * orthogonal.y,
+								startPoint.z);
+						}
+						else if (this->GeomSettings()->areEqual(basisCurveDirection.y, 0.0)) {
+							carve::geom::vector<2> startPoint2D = carve::geom::VECTOR(startPoint.x, startPoint.z);
+							carve::geom::vector<2> endPoint2D = carve::geom::VECTOR(endPoint.x, endPoint.z);
+
+							carve::geom::vector<2> tangent = endPoint2D - startPoint2D;
+							carve::geom::vector<2> orthogonal = carve::geom::VECTOR(-tangent.y, tangent.x);
+
+							offsetPoint = carve::geom::VECTOR(startPoint.x + distance * orthogonal.x,
+								startPoint.y,
+								startPoint.z + distance * orthogonal.y);
+						}
+						else if (this->GeomSettings()->areEqual(basisCurveDirection.x, 0.0)) {
+							carve::geom::vector<2> startPoint2D = carve::geom::VECTOR(startPoint.y, startPoint.z);
+							carve::geom::vector<2> endPoint2D = carve::geom::VECTOR(endPoint.y, endPoint.z);
+
+							carve::geom::vector<2> tangent = endPoint2D - startPoint2D;
+							carve::geom::vector<2> orthogonal = carve::geom::VECTOR(-tangent.y, tangent.x);
+
+							offsetPoint = carve::geom::VECTOR(startPoint.x,
+								startPoint.y + distance * orthogonal.x,
+								startPoint.z + distance * orthogonal.y);
+						}
+						else {
+							throw oip::UnhandledException(offsetCurve2D);
+						}
+
+						offsetCurve2DPoints.push_back(offsetPoint);
+					}
+
+					carve::geom::vector<3> lastStartPoint = basisCurvePoints[basisCurvePoints.size()-1];
+					carve::geom::vector<3> lastEndPoint = basisCurvePoints[basisCurvePoints.size() - 2];
+					carve::geom::vector<2> lastStartPoint2D = carve::geom::VECTOR(lastStartPoint.x, lastStartPoint.y);
+					carve::geom::vector<2> lastEndPoint2D = carve::geom::VECTOR(lastEndPoint.x, lastEndPoint.y);
+					carve::geom::vector<2> lastTangent = lastEndPoint2D - lastStartPoint2D;
+					carve::geom::vector<2> lastOrthogonal = carve::geom::VECTOR(-lastTangent.y, lastTangent.x);
+
+					carve::geom::vector<3> lastOffsetPoint = carve::geom::VECTOR(lastStartPoint.x - distance * lastOrthogonal.x,
+						lastStartPoint.y - distance * lastOrthogonal.y,
+						lastStartPoint.z);

```

Comment:
Might need to be changed

Reply:
Possible solution: [e4565ca](https://github.com/tumcms/Open-Infra-Platform/pull/558/commits/e4565ca74f0e961fa1a01a283695f6f60e536c07)